wicketd-client/src/lib.rs

```diff
@@ -4,6 +4,27 @@
 
 //! Interface for making API requests to wicketd
 
-use omicron_common::generate_logging_api;
-
-generate_logging_api!("../openapi/wicketd.json");
+progenitor::generate_api!(
+    spec = "../openapi/wicketd.json",
+    inner_type = slog::Logger,
+    pre_hook = (|log: &slog::Logger, request: &reqwest::Request| {
+        slog::debug!(log, "client request";
+            "method" => %request.method(),
+            "uri" => %request.url(),
+            "body" => ?&request.body(),
+        );
+    }),
+    post_hook = (|log: &slog::Logger, result: &Result<_, _>| {
+        slog::debug!(log, "client response"; "result" => ?result);
+    }),
+    derives = [schemars::JsonSchema],
+    patch =
+        {
+        SpIdentifier = { derives = [Copy, PartialEq, Eq, PartialOrd, Ord] },
+        SpState = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpComponentInfo= { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpIgnition= { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        SpInventory = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+        RackV1Inventory = { derives = [ PartialEq, Eq, PartialOrd, Ord] },
+    }
```

Comment:
Sweet!

Reply:
@ahl did an awesome job here. It just worked!